### ISSUE TITLE
Crash for 30 Min recording on Moto G3 (2015)

### ISSUE THREAD

**Issue by abhijeetpathak** [icechen1/BlackBox#3]
Crashes when I select 30 mins and press record.

It's trying to allocate around 151MB in CircularByteBuffer.

Following is the crash:

```
02-20 17:24:07.894    5018-5125/icechen1.com.blackbox E/AndroidRuntime﹕ FATAL EXCEPTION: Thread-78157
    Process: icechen1.com.blackbox, PID: 5018
    java.lang.OutOfMemoryError: Failed to allocate a 158760012 byte allocation with 6858288 free bytes and 86MB until OOM
            at icechen1.com.blackbox.audio.CircularByteBuffer.<init>(CircularByteBuffer.java:21)
            at icechen1.com.blackbox.audio.AudioBufferManager.run(AudioBufferManager.java:96)
```

I'm also looking into fixing the solution.

**Comment by yucombinator** [comment]
Thanks for the report! I made some changes with how memory is allocated, so maybe give it a try (https://github.com/icechen1/BlackBox/commit/e86c2d03a571aac9e5967701ab496f756092b091)? The gist is that the app will try lower sample rates if we are unable to acquire that much memory (lower sample rate = smaller amount of required memory). It might still fail in extreme cases, though, so I also improved error reporting so it does not simply crash.
